src/app/space/create/deployments/services/deployments.service.ts

```diff
@@ -259,6 +293,47 @@ export class DeploymentsService implements OnDestroy {
       );
   }
 
+  getDeploymentTimeConstrainedStats(spaceId: string, applicationId: string, environmentName: string, elapsedTime: number, endTime: number = Date.now()): Observable<TimeConstrainedStats> {
+    const startTime = endTime - elapsedTime;
+    const series = this.getTimeConstrainedTimeseriesData(spaceId, applicationId, environmentName, startTime, endTime);
+    // Process CPU Stats
+    const coresSeries: Observable<CoresSeries[]> = series.map((t: MultiTimeseriesData) => t.cores);
+    const cpuQuota = this.getEnvironmentCpuStat(spaceId, environmentName).map((stat: CpuStat) => stat.quota);
+    let cpuStats = Observable.combineLatest(coresSeries, cpuQuota, (coresSeries: CoresSeries[], cpuQuota: number) => {
+      let arr = [];
+      for (let i = 0; i < coresSeries.length; i++) {
```

Comment:
Why are you looping over an Observable? Usually you Subscribe to them.

Reply:
He's looping over an array emitted by an Observable, but it is a little confusing because the name is shadowed.

Reply:
it appears to a variable with the type of Observable.

Reply:
I'm trying to combine the cores series observable and quota observable similar to how it is done in [getDeploymentCpuStat](https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/deployments/services/deployments.service.ts#L282) [2]. In that instance there is only one cores series and one quota, so it's a quick combination and return. With the response that /statseries returns we get an array of coreseries, and combining each index with the quota isn't as straightforward? The loop accomplishes what [2] does for however many data points the api response returns (in this case, potentially 15 points). Since this cpuStats observable gets combined at the end of the function with the memory and network observable, does it make sense to manipulate the data while it's still an observable? Or should it be subscribed to for formatting, and then combined as an observable for the function return value?

[2] https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/space/create/deployments/services/deployments.service.ts#L282

Reply:
@joshuawilson it's not - there are two variables and the array one shadows the Observable one.

`const coresSeries: Observable<CoresSeries[]> = series.map((t: MultiTimeseriesData) => t.cores);` is the first, outer declaration, but the second usage that is iterated over as an array comes from the lambda parameter: `let cpuStats = Observable.combineLatest(coresSeries, cpuQuota, (*coresSeries*: CoresSeries[], cpuQuota: number) => {`

Reply:
thanks for pointing it out